Setup global environment params inside the OptionsManager

christge-sft opened the pull request:
This is an attempt to solve [issue-3971](https://github.com/cvmfs/cvmfs/issues/3971)

What is happening here basically is we're transferring the `SetupGlobalEnvironmentParams()` inside the `OptionsManager`.

The PR is marked as draft since probably some further cleanup is needed (Now the same functionality is provided from the`OptionsManager` and the `Filesystem`).

To test that the environment is setup properly, we provide a temporary `/etc/cvmfs/default.local` that reports if `CVMFS_VERSION` and `CVMFS_ARCH` are properly set.

vvolkl committed:
prerelease 2.13.2~pre2

vvolkl committed:
[rpm] suse: ~preN should not be in /usr/share/doc/<version> (#3937)

vvolkl committed:
[deb] remove debian_revision from package version (#3938)

This is only allowed for non-native packages. It's worth thinking about
switching the format to quilt in the future:
https://wiki.debian.org/Projects/DebSrc3.0

andrey-utkin committed:
Test 055-ownership: fix CVMFS_USER inference (#3921)

The test sources only /etc/cvmfs/default.conf, while it should consult
all the parameter files as described in mount/default.conf comment, or
use `cvmfs_config showconfig`.

vvolkl committed:
cmake: add BUILTIN_EXTERNALS_{LIST|EXCLUDE} options (#3940)

Allows some granularity for which vendored libs are being built. The
main purpose is to allow Arch Linux packagers to remove a patch (which I
broke because I moved the bootstrap script).

vvolkl committed:
telemetry: make delta values optional (#3939)

vvolkl committed:
Re-add fuse3 dependency to server until fstab is fixed (#3943)

Fixes https://github.com/cvmfs/cvmfs/issues/3901

vvolkl committed:
update changelog

vvolkl committed:
`cvmfs_talk metrics prometheus`  (#3944)

This adds a talk command that outputs a lot of information in a format
that can be ingested by Prometheus. The motivation is to speed up the
script in https://github.com/cvmfs-contrib/prometheus-cvmfs , which was
very slow due to the overhead of calling cvmfs_talk over and over for
each metric. Having all metrics returned by one command reduces the
runtime of the script from something like 13s to 2.5s second for 20
mounted repos.

A possible future optimization could be to offload some work, like
constructing the string, from talk.cc to cvmfs_talk.cc, and only send
the raw values in a suitable format over the socket.

vvolkl committed:
bump version to 2.13.2

vvolkl committed:
update changelog

vvolkl committed:
macos: fixes for 2.13.2 (#3945)

Fix forgotten line in packaging script, set macFUSE to be default build option

andrey-utkin committed:
Test 084-premounted: rely on run.sh for workdir cleaning (#3924)

This test's code has an immediate issue that upon test failure, workdir
is still wiped, preventing immediate analysis.

test/run.sh already takes care of creation of workdir (the test function
is launched with it as current working dir), and similarly of removing
of it upon successful completion.

We still need a cleanup procedure to unmount our mount. This commit
prevents multiple executions of the cleanup routine.

andrey-utkin committed:
test/run.sh: support non-regular files as logfile, e.g. /dev/stdout (#3922)

Before, `wc -l` commands which apparently calculate and log the log line
numbers, stalled the execution in such case.

andrey-utkin committed:
Fixes to 500-concurrent-publishing test (#3942)

vvolkl committed:
Deduplicate integration test numbers (#3941)

DrDaveD committed:
Remove cvmfs-server fuse3 requirement from rpm packaging (#3946)

This was unnecessarily added in #3943. It was trying to ensure that
`mount.fuse` was installed for release manager machines, but the `fuse3`
package on RHEL systems doesn't even have that, it has `mount.fuse3`
instead, and the package that does have it (`fuse`) is still required by
the cvmfs package. The problem fixed by #3943 was a debian-only problem
and solution.

DrDaveD committed:
Fix updating of last_gc when no gc is run under gc -a without dry run (#3947)

Fix mistake from #3895 which prevents the "last_gc" time from being
updated in .cvmfs_status.json when gc is skipped. The check for dry_run
was reversed. This bug causes cvmfs-servermon to complain if garbage
collection is skipped for a time, default 30 days.

DrDaveD committed:
Really fix updating of last_gc when no gc is run under gc -a without dry run (#3950)

It turns out that the "fix" in #3947 didn't actually change any
behavior. That's what I get for changing it after testing. This is what
I meant to submit, after having tested it with `$dry_run -ne 1` but
thinking it would be slightly better to go with comparing `$dry_run`
with `0` instead.

vvolkl committed:
Make googletest dependency optional (#3949)

Dave reported an issue while rebuilding 2.13.2 from the source package
using koji. Because that build is done without network access, the build
failed with a missing googletest dependency due to
https://github.com/cvmfs/cvmfs/pull/3874

We've added a packaging patch to include googletest in the src package
in https://github.com/cvmfs/cvmfs/tree/cvmfs-2.13.2-2 , but there's
really no good reason for our unittest library to be a required
dependency. This PR makes it so that googletest is only needed when the
unittests are actually built, by adding a forwarding header
`duplex_testing.h` that is included instead of `gtest_prod.h`. It also
makes the cvmfs rpm spec file skip the unittest package by default. It
can still be built with `rpmbuild --define="build_testing 1"`. I don't
really get the need to package the unittests though, and might remove
this alltogether in the future ( this PR already removes the unittests
from debian packaging ). I find that it's more important to have the
option to run the unittests during the package build, failing early
before even creating the package. Before we can enable this by default,
the T_Namespace tests should be looked at or disabled though, because
they make trouble when testing in a container.

Just in case, it also allows for the googletest source to be downloaded
to the `externals` directory before cmake is invoked. This can serve as
a template for the actually required, vendored dependencies that don't
have to be checked into the repo with a CMake configured that way.

vvolkl committed:
CMake improvements (#3951)

* Switch on creation of `compile_commands.json` by default. Not
expensive, and useful for developers using an LSP.
* Skip checking of system headers
* rename external build and install dirs to include platform
information, p.ex. `externals_install.x86_64.ubuntu22.04`. Useful for
caching of those dirs outside of the repo root, when doing container
builds for example.

vvolkl committed:
[rpm] Fix fuse3 libraries on suse (#3954)

The fuse3 libraries on SLES have a different package name. It seems like
cvmfs using fuse3 on SUSE has never been tried before, because it seems
like this has always been broken, and just surfaced when we switched to
fuse3 by default recently.

I'll make a 2.13.2-3 build  for SUSE.

vvolkl committed:
Fix wrong variable names in cvmfs_server_gc, faulty cherry-pick of (#3575) (#3956)

https://github.com/cvmfs/cvmfs/pull/3895 renamed `all_collectable` to
`all_collected`, but then, when https://github.com/cvmfs/cvmfs/pull/3575
was cherry-picked back on devel, it added a block using the old variable
names.

Should have been caught by tests, so that's a todo. Maybe we also do
need the strict mode for bash.

vvolkl committed:
cmake: better externals dir naming, re-use legacy externals folders (#3955)

>It's probably not wrong to add some logic to keep using legacy
`externals_install` directories if they exist. Otherwise a new set of
externals will be built in the new directory, but the build directory
will keep using the old one, that can be a bit confusing. Will still add
a patch

_Originally posted by @vvolkl in
https://github.com/cvmfs/cvmfs/issues/3951#issuecomment-3144853285_

This is the patch. Also uses PLATFORM_ID instead of the unnecessarily
specific DISTRO_ID where available, so the name is
`externals_install.x86_64.el9/` instead of
`externals_install.x86_64.almalinux9.6/`

vvolkl committed:
fix mistyped variable name in test 682 (#3957)

vvolkl committed:
ci: script to list/install system dependencies (#3963)

Convenient, especially for for new cloud tests

vvolkl committed:
Workarounds for macOS CI (#3968)

* Fix bootstrap.sh to work with BSD sed on macOS
* Update runner image to macOS 15
* Use sqlite3 from system
* Disable two failing tests ( to be investigated )

andrey-utkin committed:
Update python shebangs to python3 (#3904)

This was seen in client tests:

-- Testing Cancel file system request (Mon Jul 7 13:03:04 UTC 2025 /
test number 096)
	restarting autofs...
	using systemd
	# Created by CVMFS integration tests
	CVMFS_SERVER_APACHE_RELOAD_IS_RESTART=true
	*** install a desaster cleanup
	*** mount repository
	*** run a silent HTTP server
Calling /home/sftnight/cvmfs/test/common/mock_services/silent_socket.py
TCP 8019
	silent server started with PID 69599
	env: 'python': No such file or directory

DrDaveD committed:
Treat an ENOENT from repository unmount as normal (#3970)

- Fixes #3969

DrDaveD committed:
Change geoip database to openhtc source by default (#3967)

This maintains the maxmind geoip database as an option for the stratum 1
geo api but changes the default to reading via
https://geoipdb.openhtc.io which currently redirects to the download
area of the [cvmfs-contrib/config-repo geoipdb tag release
assets](https://github.com/cvmfs-contrib/config-repo/releases/tag/geoipdb).

---------

Co-authored-by: Valentin Volkl <valentin.volkl@cern.ch>

vvolkl committed:
Include date and hostname in bugreport tarball filename (#3962)

A quality of life improvement for developers, who otherwise need to
manually rename those files on download to keep track of them.

vvolkl committed:
[rpm] silence a mkdir warning when /cvmfs already exists (#3972)

When cvmfs was previously installed from source, or /cvmfs/ exists due
to some other reason, mkdir would print a non-fatal error message that
isn't necessary.

DrDaveD committed:
Fix intermittent failures in gc -a test (#3975)

There were three problems with the tests for gc -a added in #3895 that
this fixes. This also restores the ownership on /var/log/cvmfs that the
test changes, and adds the snapshots.log and gc.log to the test log for
better debugging.

- Fixes #3974

DrDaveD committed:
Update geo api for either mmdb name (#3982)

Unfortunately I didn't fully test the new geo db download in #3967, and
we apparently didn't notice that the cloudtest 614-geoservice would have
failed. This fixes that situation, including removing the alternate
database name when switching between the two different types.

Test 614-geoservice takes about 3 minutes so it isn't exactly "quick"
but I wonder if we should include it in the quick tests anyway so it
will always run. Alternatively, maybe the Jenkins cloudtests
configuration can be changed to use the new geo database.

vvolkl committed:
Replace AUTHORS with CITATION.cff (#3965)

No update to the content. The main reason for this change, apart from
CITATION.cff being the commonly accepted format on github and Zenodo
nowadays, is to aid the creation of new entries in Zenodo. Previously
this would just scan the git history for authors, but with a
CITATION.cff it honors our author list.

Unfortunately citation-file-format v1.3.0 is not released yet. It would
have included more fine-grained contributor roles besides Authors. Until
it is available I leave the further acknowledgements as a comment.

vvolkl committed:
ci: fixes for macos (#3989)

* bump fuse-t version
* workaround fuse header includes with CPATH
* enable fuse-t xattrs that were disabled by default
* use zlib from system

mharvey-jt committed:
correctly enable swissknife ingestsql; add CVMFS_MAX_TTL_SECS (#3980)

* the ingestsql feature was not correctly enabled in the
cvmfs_swissknife binary
* the -B option required implementation
* add the CVMFS_MAX_TTL_SECS option to set a TTL < 60 secs.

---------

Co-authored-by: Matt Harvey <mharvey@fpif1-vcsl1.w2k.jumptrading.com>
Co-authored-by: Valentin Volkl <valentin.volkl@cern.ch>

andrey-utkin committed:
catalog_rw: hold lock for VacuumDatabaseIfNecessary() (#3990)

Observed SQLite errors
"cannot start a transaction within a transaction". The same catalog's
database gets modified and vacuumed at the same time.

```
 Thread 2 (Thread 9156.9163 (cvmfs_swissknife_debug)):
 #0  0x000076a37a5ac728 in sqlite3_step () from /lib64/libsqlite3.so.0
 #1  0x00000000004e87cb in sqlite::Sql::Execute (this=0x5b09d10) at /usr/local/src/cvmfs/cvmfs/sql.cc:43
 #2  0x0000000000441410 in sqlite::Database<catalog::CatalogDatabase>::BeginTransaction (this=0x5b09680) at /usr/local/src/cvmfs/cvmfs/sql_impl.h:269
 #3  0x000000000044332e in catalog::WritableCatalog::Transaction (this=0x5b43f60) at /usr/local/src/cvmfs/cvmfs/catalog_rw.cc:75
 #4  0x000000000042bf54 in catalog::WritableCatalog::SetDirty (this=0x5b43f60) at /usr/local/src/cvmfs/cvmfs/catalog_rw.h:164
 #5  0x0000000000445867 in catalog::WritableCatalog::UpdateNestedCatalog (this=0x5b43f60, path="/dir/inner_dir5/inner_dir", hash=..., size=57344, child_counters=...) at /usr/local/src/cvmfs/cvmfs/catalog_rw.cc:596
 #6  0x000000000042aae9 in catalog::WritableCatalogManager::SingleCatalogUploadCallback (this=0x7fff5448d920, result=...) at /usr/local/src/cvmfs/cvmfs/catalog_mgr_rw.cc:1674
 #7  0x0000000000439812 in BoundCallback<upload::SpoolerResult, catalog::WritableCatalogManager>::operator() (this=0x5a4f1a0, value=...) at /usr/local/src/cvmfs/cvmfs/util/async.h:91
 #8  0x000000000046ee5e in Observable<upload::SpoolerResult>::NotifyListeners (this=0x59563f0, parameter=...) at /usr/local/src/cvmfs/cvmfs/util/concurrency_impl.h:174
 #9  0x00000000005c35b5 in upload::Spooler::ProcessingCallback (this=0x59563f0, data=...) at /usr/local/src/cvmfs/cvmfs/upload.cc:133
 #10 0x00000000005c3d90 in BoundCallback<upload::SpoolerResult, upload::Spooler>::operator() (this=0x5969930, value=...) at /usr/local/src/cvmfs/cvmfs/util/async.h:91
 #11 0x000000000046ee5e in Observable<upload::SpoolerResult>::NotifyListeners (this=0x596b960, parameter=...) at /usr/local/src/cvmfs/cvmfs/util/concurrency_impl.h:174
 #12 0x000000000046c1bf in IngestionPipeline::OnFileProcessed (this=0x596b960, spooler_result=...) at /usr/local/src/cvmfs/cvmfs/ingestion/pipeline.cc:115
 #13 0x0000000000474198 in BoundCallback<upload::SpoolerResult, IngestionPipeline>::operator() (this=0x596ad30, value=...) at /usr/local/src/cvmfs/cvmfs/util/async.h:91
 #14 0x000000000046ee5e in Observable<upload::SpoolerResult>::NotifyListeners (this=0x596d550, parameter=...) at /usr/local/src/cvmfs/cvmfs/util/concurrency_impl.h:174
 #15 0x0000000000478be7 in TaskRegister::Process (this=0x596d540, file_item=0x5b484e0) at /usr/local/src/cvmfs/cvmfs/ingestion/task_register.cc:30
 #16 0x0000000000470c31 in TubeConsumer<FileItem>::MainConsumer (data=0x596d540) at /usr/local/src/cvmfs/cvmfs/ingestion/task.h:56
 #17 0x000076a379ab1f54 in start_thread () from /lib64/libc.so.6
 #18 0x000076a379b35154 in clone () from /lib64/libc.so.6

 Thread 1 (Thread 9156.9156 (cvmfs_swissknife_debug)):
 #0  0x000076a379aba462 in __syscall_cancel_arch () from /lib64/libc.so.6
 #1  0x000076a379aae75c in __internal_syscall_cancel () from /lib64/libc.so.6
 #2  0x000076a379aae7a4 in __syscall_cancel () from /lib64/libc.so.6
 #3  0x000076a379b0b58c in pread64 () from /lib64/libc.so.6
 #4  0x000076a37a56925a in ?? () from /lib64/libsqlite3.so.0
 #5  0x000076a37a69aad3 in ?? () from /lib64/libsqlite3.so.0
 #6  0x000076a37a69afaa in ?? () from /lib64/libsqlite3.so.0
 #7  0x000076a37a59584e in sqlite3_backup_step () from /lib64/libsqlite3.so.0
 #8  0x000076a37a60a694 in ?? () from /lib64/libsqlite3.so.0
 #9  0x000076a37a5a3571 in ?? () from /lib64/libsqlite3.so.0
 #10 0x000076a37a5acae4 in sqlite3_step () from /lib64/libsqlite3.so.0
 #11 0x00000000004e87cb in sqlite::Sql::Execute (this=0x7fff5448c750) at /usr/local/src/cvmfs/cvmfs/sql.cc:43
 #12 0x0000000000446d13 in sqlite::Database<catalog::CatalogDatabase>::Vacuum (this=0x5b09680) at /usr/local/src/cvmfs/cvmfs/sql_impl.h:418
 #13 0x0000000000446652 in catalog::WritableCatalog::VacuumDatabaseIfNecessary (this=0x5b43f60) at /usr/local/src/cvmfs/cvmfs/catalog_rw.cc:767
 #14 0x000000000042913e in catalog::WritableCatalogManager::FinalizeCatalog (this=0x7fff5448d920, catalog=0x5b43f60, stop_for_tweaks=false) at /usr/local/src/cvmfs/cvmfs/catalog_mgr_rw.cc:1317
 #15 0x000000000042a6f4 in catalog::WritableCatalogManager::ScheduleReadyCatalogs (this=0x7fff5448d920) at /usr/local/src/cvmfs/cvmfs/catalog_mgr_rw.cc:1632
 #16 0x000000000052e80a in swissknife::IngestSQL::do_additions (this=0x59507a0, all_dirs=std::map with 18 elements = {...}, all_files=std::map with 10 elements = {...}, all_symlinks=std::map with 10 elements = {...}, lease_path="", catalog_manager=...) at /usr/local/src/cvmfs/cvmfs/swissknife_ingestsql.cc:1133
 #17 0x000000000052e007 in swissknife::IngestSQL::process_sqlite (this=0x59507a0, dbs=std::vector of length 1, capacity 1 = {...}, catalog_manager=..., allow_additions=true, allow_deletions=true, lease_path="", additional_prefix="") at /usr/local/src/cvmfs/cvmfs/swissknife_ingestsql.cc:1038
 #18 0x000000000052d4f4 in swissknife::IngestSQL::Main (this=0x59507a0, args=std::map with 12 elements = {...}) at /usr/local/src/cvmfs/cvmfs/swissknife_ingestsql.cc:889
 #19 0x0000000000553a40 in main (argc=24, argv=0x7fff5448f1a8) at /usr/local/src/cvmfs/cvmfs/swissknife_main.cc:189
 (rr) thread 1
 [Switching to thread 1 (Thread 9156.9156)]
 #0  0x000076a379aba462 in __syscall_cancel_arch () from /lib64/libc.so.6
 (rr) frame 11
 #11 0x00000000004e87cb in sqlite::Sql::Execute (this=0x7fff5448c750) at /usr/local/src/cvmfs/cvmfs/sql.cc:43
 43        last_error_code_ = sqlite3_step(statement_);
 (rr) print *this
 $53 = (sqlite::Sql) {
   _vptr.Sql = 0x66cfd0 <vtable for sqlite::Sql+16>,
   database_ = 0x5b0a050,
   statement_ = 0x5b34190,
   query_string_ = 0x0,
   last_error_code_ = 0
 }
 (rr) print database_
 $54 = (sqlite3 *) 0x5b0a050
 (rr) thread 2
 [Switching to thread 2 (Thread 9156.9163)]
 #0  0x000076a37a5ac728 in sqlite3_step () from /lib64/libsqlite3.so.0
 (rr) frame 2
 #2  0x0000000000441410 in sqlite::Database<catalog::CatalogDatabase>::BeginTransaction (this=0x5b09680) at /usr/local/src/cvmfs/cvmfs/sql_impl.h:269
 269       return begin_transaction_->Execute() && begin_transaction_->Reset();
 (rr) print database_
 $55 = {
   sqlite_db = 0x5b0a050,
   lookaside_buffer = 0x0,
   db_file_guard = {
     <SingleCopy> = {<No data fields>},
     members of UnlinkGuard:
     path_ = "/tmp/cvmfs_rsync_swissknife769119102/catalog.JTKTzC",
     enabled_ = true
   },
   delegate_ = 0x5b09680
 }
 (rr)
```

christge-sft committed:
set CVMFS_VERSION and CVMFS_ARCH when setting default options

christge-sft committed:
add an integration test for environment setting

vvolkl commented on test/src/108-global-config/main:
```diff
@@ -0,0 +1,51 @@
+
+  sudo rm -rf $logfile
+
+#  cvmfs_mount "atlas.cern.ch,atlas-condb.cern.ch" \
```
Clean up commented code

vvolkl commented on test/src/108-global-config/main:
```diff
@@ -0,0 +1,51 @@
+    echo "config doesn't exist"
+  fi
+
+  sudo rm -rf $logfile
```
Move to a cleanup function that can be trapped

vvolkl commented on test/src/108-global-config/main:
```diff
@@ -0,0 +1,51 @@
+cvmfs_run_test() {
+  logfile=$1
+
+  local logfile=/tmp/version_env_test.log
```
Should eventually use a workspace directory that can be set from run.sh - but that's to be fixed for a  lot of integration tests,  fine for now.

christge-sft replied:
For that with the new version I'm using the logfile provided through `$1`. So it's the responsibility of the wrapper to setup a proper logging path.

vvolkl commented on test/src/108-global-config/main:
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+cvmfs_test_name="Global Configuration Variables"
```
Not sure about the naming. I think I prefer "System Configuration Variables"

vvolkl commented on test/src/108-global-config/default.local:
```diff
@@ -0,0 +1,10 @@
+CVMFS_REPOSITORIES=atlas-condb.cern.ch
+CVMFS_HTTP_PROXY="http://ca-proxy.cern.ch:3128"
```
This shouldn't be hardcoded, we need to use CVMFS_TEST_PROXY in order to be able to run the integration tests outside the CERN network. That's done in the common setup, I think it's better to just append to the default.local instead of overwriting it in the test.

christge-sft replied:
I adjusted the approach on that: `CVMFS_TEST_PROXY` is used now.
I think that having a very minimal default.local makes the test more readable. Although if you think that appending on the current configuration is better we can do that. Although, there should be a minimal configuration in case default.local is not  there. 

I'm resolving the trhead for now and we're opening it again if necessary.

vvolkl commented on cvmfs/options.cc:
```diff
@@ -277,13 +277,30 @@ bool OptionsManager::HasConfigRepository(const string &fqrn,
 }
 
+void OptionsManager::SetupGlobalEnvironmentParams() {
+  setenv("CVMFS_ARCH", GetArch().c_str(), 1 /* overwrite */);
```
I think it may be better to use the options_mgr SetValue function here, so these variables are also visible when doing `cvmfs_talk parameters`

vvolkl reviewed (changes requested):
Looks good, thanks! Some minor comments inline. Also, I'm not sure what happens when CVMFS is upgraded - will the VERSION variable reflect that? May need to add something to the reload code to make sure it does.

christge-sft committed:
improve test name

christge-sft committed:
trap clean_up on exit

christge-sft committed:
provide as a proxy the CVMFS_TEST_PROXY

christge-sft committed:
use SetEnv instead of setenv, so that the env vars are visible to cvmfs_talk

vvolkl commented on test/src/108-global-config/main:
```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+set -e
+cvmfs_test_name="System Configuration Variables"
```
```suggestion
cvmfs_test_name="System Configuration Variables"
cvmfs_test_suites="quick"
```

christge-sft committed:
Add 108 to the quick suit

Co-authored-by: Valentin Volkl <valentin.volkl@cern.ch>

DrDaveD reviewed (changes requested):
Please also remove the `FileSystem::SetupGlobalEnvironmentParams` function from cvmfs/mountpoint.cc as extended in #3934.

christge-sft committed:
Setup CVMFS_VERSION and CVMFS_VERSION_NUMERIC in CMakeLists.txt

christge-sft committed:
Add CVMFS_VERSION and CVMFS_VERSION_NUMERIC env vars for config

christge-sft committed:
Revert "Add CVMFS_VERSION and CVMFS_VERSION_NUMERIC env vars for config"

This reverts commit 621fe17d28de72953bfd0c8ca837a0ab6102ac87.

christge-sft commented:
> Please also remove the `FileSystem::SetupGlobalEnvironmentParams` function from cvmfs/mountpoint.cc as extended in #3934.

Done! I split the original commit into two. I kept the part where `CVMFS_VERSION` and `CVMFS_VERSION_NUMERIC` are set in `CMakeLists.txt` and I reverted the rest.

DrDaveD reviewed (changes requested):
Somehow a lot of irrelevant commits have been pulled in, touching many irrelevant files.  Please clean it up.  Perhaps you'll need to recreate your branch from scratch and cherry-pick just your changes.

I see you are using your own devel branch for this.  That can be OK but pull requests usually work most smoothly when you create a new branch for them and then delete them after the PR is merged and you have updated your devel branch from upstream.

christge-sft commented:
> Somehow a lot of irrelevant commits have been pulled in, touching many irrelevant files. Please clean it up. Perhaps you'll need to recreate your branch from scratch and cherry-pick just your changes.

The reason for that is that the original commit was doing both (setting `CVMFS_VERSION` and `CVMFS_VERSION_NUMERIC` in CMakeLists.txt and the setenv inside `FileSystem::SetupGlobalEnvironmentParams`.

So this is the commit I split, and this is why you see the history after that splitting changed. But the commits themselves are not affected at all.
> 
> I see you are using your own devel branch for this. That can be OK but pull requests usually work most smoothly when you create a new branch for them and then delete them after the PR is merged and you have updated your devel branch from upstream.

Let's decide on the final version, and I can create a feature branch on my fork to issue the PR from. Although, my fork is mostly (if not only) used by myself, so it's not an issue to do a hard reset after the PR is accepted.

DrDaveD commented:
That may be the explanation but as this is I can't see what is actually changed so it's really unreviewable.  I also don't recommend doing a revert at all, because in addition to only wanting to undo part of it, there are other parts that are not covered.  Instead, please remove `FileSystem::SetupGlobalEnvironmentParams` and anywhere it is called.
